src/main/java/io/github/moulberry/notenoughupdates/overlays/FarmingSkillOverlay.java

```diff
@@ -125,7 +125,7 @@ private int mod(int x, int y) {
 
 	private void resetCropsPerSecond() {
 		cropsPerSecondTimeStamps = new long[CPS_WINDOW_SIZE];
-		cropsPerSecondValues = new int[CPS_WINDOW_SIZE];
+		cropsPerSecondValues = new float[CPS_WINDOW_SIZE];
```

Comment:
floats cant really handle the precision here. A regular float is only accurate up to 23 bits, while even an int is accurate up to 31 bits. This means that if a hoe where to reach int max this cropsPerSecondValue would only ever jump up in increments of 256 (2^(31-23). Floats would be enough if we only store the delta of some start value, but i think we store the actual counter in here. (If i am wrong just ignore this comment)